tdrs-backend/tdpservice/parsers/parse.py

```diff
@@ -36,11 +36,11 @@ def parse_datafile(datafile, dfs):
         bulk_create_errors({1: header_errors}, 1, flush=True)
         return errors
 
-    # TODO: write a test for this line
+    cat4_error_generator = util.make_generate_parser_error(datafile, None)
```

Comment:
for duplicate errors, I think it makes sense to add line number (row_number) to the error

Reply:
We can't add that up here. If we called `util. make_generate_parser_error` every time we had a cat4 error then we could add that info. But this function lives the entire lifecycle of the file being parsed. Would you think it better to move it down into the respective classes?